### PR TITLE
Sidebar: Fix e2e tests

### LINKED ISSUE #10812: Sidebar: e2e/percy tests
## Bug Description

PR should be based on/merged to the `sidebar` branch.

This ticket is to address the broken e2e tests from moving the inspector pane into the design pane (one sidebar vs 2). 

## Expected Behaviour

Passing test pipeline for sidebar branch. 

## Steps to Reproduce

Probably the easiest way to really see what's going on is make a draft pr for the sidebar to main and run e2e tests.

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by samwhale**
## Context

Helping out prometheus with the bigger than expected UI Simplification work.

Failing karma tests will be addressed in a different PR.

## Summary

Fix failing e2e tests in the `sidebar` branch.

## Relevant Technical Choices

n/a

## To-do

n/a

## User-facing changes

n/a

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Tests should pass

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10812

### CODE REVIEW

File: packages/e2e-test-utils/src/publishStory.js [hunk 1]
```diff
@@ -21,19 +21,19 @@
  * @return {Promise<void>}
  */
 async function publishStory(dismiss = true) {
-  await expect(page).toClick('button', { text: 'Publish' });
+  await expect(page).toClick('button', { text: /^Publish$/ });
```
Comment (samwhale) [inline packages/e2e-test-utils/src/publishStory.js:24]: New 'publishing' panel was added in the style (design) pane, so this was matching the button in that panel.

File: includes/data/stories/demo.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
       "elements": [
         {
           "opacity": 100,
-          "flip": [],
+          "flip": { "vertical": false, "horizontal": false },
```
Comment (samwhale) [inline includes/data/stories/demo.json:8]: Fixes console error in demo story e2e test. `flip` must be an object.

File: packages/e2e-tests/src/specs/editor/media/insertWebMVideo.js [hunk 3]
```diff
@@ -38,6 +38,9 @@ describe('Inserting WebM Video', () => {
   });
 
   async function openA11yPanel() {
+    // open style pane
+    await expect(page).toClick('li', { text: /^Style$/ });
```
Comment (samwhale) [inline packages/e2e-tests/src/specs/editor/media/insertWebMVideo.js:42]: New sidebar structure doesn't show the design pane (now called the style pane) all the time like before. To expose the panels, the user must click over to it.

File: packages/wp-dashboard/src/api/story.js [hunk 4]
```diff
@@ -160,6 +160,15 @@ export const createStoryFromTemplate = async (config, template) => {
...
         id: 0,
+        url: '',
+      },
+      title: '',
+      publisherLogo: {
+        url: '',
+      },
+      pages: [],
+      metadata: {
+        publisher: '',
```
Comment (samwhale) [inline packages/wp-dashboard/src/api/story.js:171]: Fixes proptype error when selecting a template. This error was visible locally but not in CI.

https://github.com/GoogleForCreators/web-stories-wp/blob/2a2e69f9cc532d444a8f9ac90e53fd46ac815d89/packages/story-editor/src/output/story.js#L100-L121

File: packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ describe('Panel: Style Presets', () => {
       await fixture.editor.library.textTab.click();
 
       // Verify that no styles are added currently.
-      const noStylesText = fixture.screen.getByText('No Styles Saved');
+      let noStylesText = fixture.screen.getByText('No Styles Saved');
       expect(noStylesText).toBeDefined();
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: Why make this a let? Why just rename the second instance of `noStylesText`?

Comment (miina) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: Hm, is this a merging issue? For some reason, `// Verify that no styles are added currently.` is happening twice here (here and below), there's no reason to check for it twice.

**Review by miina — APPROVED** [review]
The e2e tests pass 🎉

File: packages/wp-dashboard/src/api/story.js (hunk 4 above)
Comment (swissspidy) [reply packages/wp-dashboard/src/api/story.js:171]: Did this error occur when running e2e tests?

> This error was visible locally but not in CI.

FYI, prop types errors are only displayed when building in development mode. E2e tests are run in production mode.

Comment (samwhale) [reply packages/wp-dashboard/src/api/story.js:171]: Strangely this did occur when running the e2e tests (locally) 🤔. However, I had a hard time tracking it down in the test so I ended up finding it while running the app locally.

The error was logged when a user clicks the `Use Template` button. The error logged briefly before the route changes to the editor, and then the console was cleared.

File: packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js (hunk 5 above)
Comment (samwhale) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: This was current in the `sidebar` branch, but is unrelated to my changes. Lint was flagging it so I updated it but it seems that my update doesn't make sense 😅 

I'll take another look at it

Comment (miina) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: I'm guessing the merging issue came from before your PR 🙂 Somehow checking for `noStylesText` got in twice.

Comment (BrittanyIRL) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: think that was a bad rebase from when i updated sidebar from main on friday, thank you for fixing it.

**Review by maxyinger — APPROVED** [review]

File: packages/wp-dashboard/src/api/story.js (hunk 4 above)
Comment (swissspidy) [reply packages/wp-dashboard/src/api/story.js:171]: > Strangely this did occur when running the e2e tests (locally)

I should clarify:

On CI, tests are run like this:

1. `npm run build`
2. `npm run test:e2e`

Locally you likely had a dev build, i.e. `npm run dev`. Hence you saw the prop type warnings locally.

Comment (samwhale) [reply packages/wp-dashboard/src/api/story.js:171]: Oh! Okay I understand now, thanks!

File: packages/e2e-tests/src/specs/wordpress/postLocking.js [hunk 6]
```diff
@@ -33,9 +33,6 @@ const percyCSS = `.dashboard-grid-item-date { display: none; }`;
 const storyTitle = 'Test post lock';
 
 describe('Post Locking', () => {
-  withExperimentalFeatures(['enablePostLocking']);
-  withPlugin('e2e-tests-post-lock-mock');
```
Comment (samwhale) [inline packages/e2e-tests/src/specs/wordpress/postLocking.js]: For some reason, running this before the `beforeAll` on line 38 made the story locked _before_ the story was even created.

Moving this to line 44 adds the `beforeAll` from `withPlugin` after the `beforeAll` on line 39. Now the test creates the story that will be tested on _before_ turning on the plugin.

This doesn't change the behavior of the test, but it does ensure that the story can be created prior to the tests.

Comment (swissspidy) [reply packages/e2e-tests/src/specs/wordpress/postLocking.js]: cc @spacedmonkey 

#10897

Comment (samwhale) [reply packages/e2e-tests/src/specs/wordpress/postLocking.js]: Oh this is on main too? Good to know - so it looks like this PR fixed all the e2e tests on `sidebar`

I can remove this change and leave it for that separate ticket.

Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/wordpress/postLocking.js]: Changes are in https://github.com/GoogleForCreators/web-stories-wp/pull/10901

Comment (samwhale) [reply packages/e2e-tests/src/specs/wordpress/postLocking.js]: Going to remove my changes here 😄

**Pull request merged**